src/statistics.rs

```diff
@@ -0,0 +1,72 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+use std::slice::Iter;
+
+use stopwatch::Stopwatch;
+
+macro_rules! iterated_enum {
+    ($name: ident ; $num: expr => [ $($thing: ident),* ] ) => {
+        #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+        pub enum $name {
+            $($thing),*
+        }
+
+        impl $name {
+            fn iter() -> Iter<'static, $name> {
+                use self::$name::*;
+                static THINGS: [$name; $num] = [ $($thing),* ];
+                THINGS.into_iter()
+            }
+        }
+    }
+}
+
+iterated_enum! {
+    SWType ; 3 => [
+        Total,
+        Loading,
+        Generating
+    ]
+}
+
+pub struct Watcher {
+    stopwatch: Rc<RefCell<Stopwatch>>,
```

Comment:
Or rather it would make sense if the stopwatch crate had marked its methods as `inline`. As it is `RefCell` could even be better. Overall this doesn't affect performance anyway...